src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -608,6 +608,62 @@ public void RegisterSymbolAction(DiagnosticAnalyzer analyzer, Action<SymbolAnaly
             SymbolAnalyzerAction analyzerAction = new SymbolAnalyzerAction(action, symbolKinds, analyzer);
             this.GetOrCreateAnalyzerActions(analyzer).AddSymbolAction(analyzerAction);
             _symbolActions = _symbolActions.Add(analyzerAction);
+
+            // The SymbolAnalyzerAction does not handle SymbolKind.Local or SymbolKind.Parameter
+            // because the compiler does not make CompilationEvents for them. As a workaround,
+            // handle them specially by registering further OperationActions (for 
+            // VariableDeclarations) or SymbolActions (for Methods) and utilize the results of
+            // those to construct the necessary SymbolAnalysisContexts.
+
+            if (symbolKinds.Contains(SymbolKind.Local))
+            {
+                RegisterOperationAction(analyzer, opContext =>
```

Comment:
> How is this preferable over actually asking the analyzer authors to register these operation actions? 

1. We do not allow analzyers to register operation actions.
2. The mapping between SymbolKind and the appropriate analyzer operations is not something that may be clear to authors.  Look at the example of SymbolKind.Parameter.  Getting called back for parameters means enumerating a whole host of cases.  The onus should not be on all authors to figure out how to do this properly.  It's just something that will lead to excess fragile code to maintain.

It seems pretty clear to me.  We have RegisterSymbolAction.  We should do what we can to make sure that that works properly for all the possible SymbolKinds that can be passed in.